integration_tests/src/main/python/fastparquet_compatibility_test.py

```diff
@@ -0,0 +1,211 @@
+        conf=rebase_write_corrected_conf
+    )
+
+    # TODO: Maybe make _assert_equal() available to compare dataframes, regardless of CPU vs GPU?
+    # For now, this compares the results of reading back the GPU-written data, via fastparquet and GPU.
+    assert_gpu_and_cpu_are_equal_collect(read_parquet(data_path), conf=rebase_write_corrected_conf)
+
+
+@pytest.mark.parametrize('column_gen', [
+    ByteGen(nullable=False),
+    ByteGen(nullable=True),
+    ShortGen(nullable=False),
+    ShortGen(nullable=True),
+    IntegerGen(nullable=False),
+    IntegerGen(nullable=True),
+    LongGen(nullable=False),
+    LongGen(nullable=True),
+    FloatGen(nullable=False),
+    FloatGen(nullable=True),
+    DoubleGen(nullable=False),
+    DoubleGen(nullable=True),
+    DecimalGen(nullable=False),
+    DecimalGen(nullable=True),
+    pytest.param(
+        StringGen(nullable=False),
+        marks=pytest.mark.xfail(reason="String columns written with fastparquet are read differently between "
+                                       "Apache Spark, and the Spark RAPIDS plugin. "
+                                       "See https://github.com/NVIDIA/spark-rapids/issues/9387.")),
+    pytest.param(
+        StringGen(nullable=True),
+        marks=pytest.mark.xfail(reason="String columns written with fastparquet are read differently between "
+                                       "Apache Spark, and the Spark RAPIDS plugin. "
+                                       "See https://github.com/NVIDIA/spark-rapids/issues/9387.")),
+    pytest.param(
+        DateGen(nullable=False,
+                start=date(year=2000, month=1, day=1),
+                end=date(year=2020, month=12, day=31)),
+        marks=pytest.mark.xfail(reason="spark_df.toPandas() problem: Dates generated in Spark can't be written "
+                                       "with fastparquet, because the dtype/encoding cannot be deduced.")),
+    pytest.param(
+        TimestampGen(nullable=False),
+        marks=pytest.mark.xfail(reason="Timestamps exceeding year=2300 are out of bounds for Pandas.")),
+    pytest.param(
+        TimestampGen(nullable=False,
+                     start=datetime(2000, 1, 1, tzinfo=timezone.utc),
+                     end=datetime(2200, 12, 31, tzinfo=timezone.utc)),
+        marks=pytest.mark.xfail(reason="spark_df.toPandas() problem: Timestamps in Spark can't be "
+                                       "converted to pandas, because of type errors. The error states: "
+                                       "\"TypeError: Casting to unit-less dtype 'datetime64' is not supported. "
+                                       "Pass e.g. 'datetime64[ns]' instead.\"")),
+    pytest.param(
+        ArrayGen(IntegerGen(nullable=False), nullable=False),
+        marks.pytest.mark.xfail(reason="spark.toPandas() problem: toPandas() converts Array columns into String. "
```

Comment:
`marks = pytest` not `marks.pytest` There were test failures

Reply:
Yikes. Late night editing. Will fix.